universal-application-tool-0.0.1/app/services/applicant/JsonPathPredicate.java

```diff
@@ -0,0 +1,22 @@
+package services.applicant;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/** Holds a JsonPath (https://github.com/json-path/JsonPath) path containing a query predicate. */
```

Comment:
design question (non-blocking): Will this hold the full JsonPath  predicate, including the question paths?

Reply:
I don't think so, at the moment I think it will just hold this string value. I added it to avoid primitive obsession and to ensure JsonPathPredicate as a first class domain concept in the system.